huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go

```diff
@@ -0,0 +1,280 @@
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+										ValidateFunc: validation.All(
+											validation.StringMatch(regexp.MustCompile(`^[A-Za-z-_][\w-.]*$`),
+												"The name can only contain letters, digits, underscores (_), "+
+													"hyphens (-) and dots (.), and cannot start with a digit or dot."),
+											validation.StringLenBetween(1, 64),
+										),
+									},
+									"value": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringLenBetween(1, 2048),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"component_ids": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func createEnvironmentVariables(client *golangsdk.ServiceClient, appId string, envSet *schema.Set) error {
+	for _, env := range envSet.List() {
+		envMap := env.(map[string]interface{})
+		variables := make([]applications.Variable, envMap["variable"].(*schema.Set).Len())
+		for i, v := range envMap["variable"].(*schema.Set).List() {
+			variable := v.(map[string]interface{})
+			variables[i] = applications.Variable{
+				Name:  variable["name"].(string),
+				Value: variable["value"].(string),
+			}
+		}
+
+		envId := envMap["id"].(string)
+		config := applications.Configuration{
+			EnvVariables: variables,
+		}
+		if _, err := applications.UpdateConfig(client, appId, envId, config); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func resourceApplicationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.ServiceStageV2Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating ServiceStage v2 client: %s", err)
+	}
+
+	desc := d.Get("description").(string)
+	opt := applications.CreateOpts{
+		Name:                d.Get("name").(string),
+		Description:         &desc,
+		EnterpriseProjectId: config.GetEnterpriseProjectID(d),
+	}
+	log.Printf("[DEBUG] The CreateOpts of ServiceStage application is: %v", opt)
+	resp, err := applications.Create(client, opt)
+	if err != nil {
+		return diag.Errorf("error creating ServiceStage application: %s", err)
+	}
+
+	d.SetId(resp.ID)
+
+	err = createEnvironmentVariables(client, d.Id(), d.Get("environment").(*schema.Set))
+	if err != nil {
+		return diag.Errorf("error creating environment variable: %s", err)
+	}
+
+	return resourceApplicationRead(ctx, d, meta)
+}
+
+func flattenEnvironmentVariables(appId string, configs []applications.ConfigResp) []map[string]interface{} {
+	// Store all variables of the same environment in the corresponding list, and generate a map with the environment ID
+	// as the key name.
+	result := make([]map[string]interface{}, 0)
+	for _, v := range configs {
+		if v.ApplicationId == appId && len(v.Configuration.EnvVariables) >= 1 {
```

Comment:
`v.ApplicationId == appId` is always true?

Reply:
yeah, and I think the length check is unnecessary too, so I removed both.